aztokenprovider/token_provider.go

```diff
@@ -164,20 +164,20 @@ func (c *managedIdentityTokenRetriever) GetAccessToken(ctx context.Context, scop
 }
 
 type clientSecretTokenRetriever struct {
-	authority    string
+	cloudConf    cloud.Configuration
 	tenantId     string
 	clientId     string
 	clientSecret string
 	credential   azcore.TokenCredential
 }
 
 func (c *clientSecretTokenRetriever) GetCacheKey() string {
-	return fmt.Sprintf("azure|clientsecret|%s|%s|%s|%s", c.authority, c.tenantId, c.clientId, hashSecret(c.clientSecret))
+	return fmt.Sprintf("azure|clientsecret|%s|%s|%s|%s", c.cloudConf, c.tenantId, c.clientId, hashSecret(c.clientSecret))
```

Comment:
This is probably not working as it should

Reply:
yes, you can do just `cloudConf.ActiveDirectoryAuthorityHost` to leave it as it was